### ISSUE TITLE
file.comment with regex mangles commented lines

### ISSUE THREAD

**Issue by notpeter** [saltstack/salt#27486]
If you use file.comment with a regex, the new line(s) in the file include the regex itself, not the original text which matched the regex.

For example:
````
disable_pam_motd:
  file.comment:
    - name: /etc/pam.d/sshd
    - regex: ^session\s+optional\s+pam_motd.so
````
Yields the following results:
````
----------
          ID: disable_pam_motd
    Function: file.comment
        Name: /etc/pam.d/sshd
      Result: False
     Comment: Expected commented lines not found
     Started: 23:33:02.628838
    Duration: 13.237 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -30,8 +30,8 @@
                   # Print the message of the day upon successful login.
                   # This includes a dynamically generated part from /run/motd.dynamic
                   # and a static (admin-editable) part from /etc/motd.
                  -session    optional     pam_motd.so  motd=/run/motd.dynamic noupdate
                  -session    optional     pam_motd.so # [1]
                  +#session\s+optional\s+pam_motd.so  motd=/run/motd.dynamic noupdate
                  +#session\s+optional\s+pam_motd.so # [1]
                   
                   # Print the status of the user's mailbox upon successful login.
                   session    optional     pam_mail.so standard noenv # [1]
````
(notice the "\s+" in the new lines instead of the original spaces)
````
# salt-call --versions-report
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: Not Installed
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
````

**Comment by jfindlay** [comment]
@notpeter, thanks for the report.  This appears to be fixed on 2015.8.0 and later.
```console
jmoney-debian salt # cat /etc/file
#session    optional     pam_motd.so  motd=/run/motd.dynamic noupdate
#session    optional     pam_motd.so # [1]
```
```saltstack
jmoney-debian salt # cat /srv/salt/comment.sls
disable_pam_motd:
  file.comment:
    - name: /etc/file
    - regex: ^session\s+optional\s+pam_motd.so
```
```yaml
jmoney-debian salt # salt jmoney-debian state.apply comment
jmoney-debian:
----------
          ID: disable_pam_motd
    Function: file.comment
        Name: /etc/file
      Result: True
     Comment: Commented lines successfully
     Started: 11:59:48.584795
    Duration: 11.39 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,2 +1,2 @@
                  -session    optional     pam_motd.so  motd=/run/motd.dynamic noupdate
                  -session    optional     pam_motd.so # [1]?
                  +#session    optional     pam_motd.so  motd=/run/motd.dynamic noupdate
                  +#session    optional     pam_motd.so # [1]?

Summary for jmoney-debian
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  11.390 ms
```
```yaml
jmoney-debian salt # salt --versions
Salt Version:
           Salt: 2015.8.0-24-gb71de75

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.5.4
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.1
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.1
```

**Comment by cachedout** [comment]
Since this appears to be fixed and we have heard no reports to the contrary, we are going to go ahead and close this one.
